pysap/extensions/transform.py

```diff
@@ -10,69 +10,376 @@
+    class_name = name.title()
+
+    # Define the trsform class parameters
+    class_parameters = {
+        "__module__": destination_module_globals["__name__"],
+        "_id":  destination_module_globals["__name__"] + "." + class_name,
+        "_pywt_name": name,
+        "_pywt_func": func
+    }
+
+    # Get the process instance associated to the function
+    destination_module_globals[class_name] = (
+        type(class_name, (PyWaveletTransformBase, ), class_parameters))
+
+
+destination_module_globals = globals()
+for family in pywt.families():
+    if family in ("gaus", "mexh", "morl", "cgau", "shan", "fbsp", "cmor"):
+        func = pywt.ContinuousWavelet
+    else:
+        func = pywt.Wavelet
+    for name in pywt.wavelist(family):
+        pywt_class_factory(func, name, destination_module_globals)
 
 
 class ISAPWaveletTransformBase(WaveletTransformBase):
     """ Define the structure that will be used to store the ISAP results.
     """
+    __family__ = "isap"
     __isap_transform_id__ = None
     __isap_name__ = None
     __is_decimated__ = None
     __isap_nb_bands__ = None
     __isap_scale_shift__ = 0
 
+    def __init__(self, nb_scale, verbose=0, dim=2, **kwargs):
+        """ Initialize the WaveletTransformBase class.
+
+        Parameters
+        ----------
+        data: ndarray
+            the input data.
+        nb_scale: int
+            the number of scale of the decomposition that includes the
+            approximation scale.
+        verbose: int, default 0
+            control the verbosity level.
+        dim: int, default 2
+            define the data dimension.
+        """
+        # ISAP Wavelet transform parameters
+        self.bands_lengths = None
+        self.bands_shapes = None
+        self.isap_transform_id = None
+        self.flatten_fct = None
+        self.unflatten_fct = None
+        self.scales_lengths = None
+        self.scales_padds = None
+        self.use_wrapping = pysparse is None
+
+        # Inheritance
+        super(ISAPWaveletTransformBase, self).__init__(
+            nb_scale, verbose=verbose, dim=dim, **kwargs)
+
+    def _init_transform(self, **kwargs):
+        """ Define the transform.
+        """
+        if not self.use_wrapping:
+            kwargs["type_of_multiresolution_transform"] = (
+                self.__isap_transform_id__)
+            kwargs["number_of_scales"] = self.nb_scale
+            if self.data_dim == 2:
+                self.trf = pysparse.MRTransform(**kwargs)
+            elif self.data_dim == 3:
+                self.trf = pysparse.MRTransform3D(**kwargs)
+            else:
+                raise NameError("Please define a correct dimension for data.")
+        else:
+            if self.data_dim == 2:
+                self.trf = None
+            else:
+                raise NameError("For {0}D, only the bindings work for "
+                                "now.".format(self.data_dim))
+
+    def _analysis(self, data, **kwargs):
```

Comment:
Should we add an "axes" parameter which could be helpful for multichannel reconstruction for example

Reply:
I  added an axes parameter that is specific to PYWT.